pkg/liveupdate/memory/memory_test.go

```diff
@@ -0,0 +1,82 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright The Kubevirt Authors
+ *
+ */
+
+package memory_test
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/api/resource"
+
+	v1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/pkg/libvmi"
+	"kubevirt.io/kubevirt/pkg/liveupdate/memory"
+)
+
+var _ = Describe("LiveUpdate Memory", func() {
+	Context("Memory", func() {
+
+		Context("Validation", func() {
+
+			DescribeTable("should reject VM creation if", func(maxGuestStr string, opts ...libvmi.Option) {
+				vmiOpts := []libvmi.Option{
+					libvmi.WithArchitecture("amd64"),
+				}
+				vmiOpts = append(vmiOpts, opts...)
+
+				vm := libvmi.NewVirtualMachine(libvmi.New(vmiOpts...))
+
+				maxGuest := resource.MustParse(maxGuestStr)
+				err := memory.ValidateLiveUpdateMemory(&vm.Spec.Template.Spec, &maxGuest)
+				Expect(err).To(HaveOccurred())
```

Comment:
We should check the content of the error otherwise we can't know if we wrote the test correctly.